src/shell/eventlogger.c

```diff
@@ -111,8 +112,12 @@ timer_cb (flux_reactor_t *r, flux_watcher_t *w, int revents, void *arg)
     flux_t *h = ev->h;
     double timeout = ev->commit_timeout;
     flux_future_t *f = NULL;
+    int flags = 0;
 
-    if (!(f = flux_kvs_commit (h, NULL, 0, batch->txn))
+    if (ev->flags & EVENTLOGGER_COMPACT)
+        flags |= FLUX_KVS_TXN_COMPACT;
+
```

Comment:
I don't want to hold this PR up any longer, but I can't think of a reason the eventlogger would *not* want to compact transactions always, so I'd say drop the `ev->flags` for now and always add `FLUX_KVS_TXN_COMPACT` flag to `flux_kvs_commit`. This simplifies the changes here, and allows the logging code to automatically take advantage of these improvements.

Reply:
sounds reasonable, let me modify really quick